<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width initial-scale=1">

  <title>the Darling lab | computational (meta)genomics</title>
  <meta name="description" content="The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://darlinglab.org/mauve/user-guide/mauvealigner.html">
  <link rel="alternate" type="application/atom+xml" title="the Darling lab | computational (meta)genomics" href="http://darlinglab.org/feed.xml" />
  <link rel="shortcut icon" href="/site32.jpg" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">the Darling lab | computational (meta)genomics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/mauve/mauve.html">Mauve</a>
          
        
          
        
          
          <a class="page-link" href="/openings/">Openings</a>
          
        
          
        
          
          <a class="page-link" href="/people/">People</a>
          
        
          
        
          
          <a class="page-link" href="/projects/">Projects</a>
          
        
          
          <a class="page-link" href="/publications/">Publications</a>
          
        
          
        
          
          <a class="page-link" href="/reviews/">Reviews</a>
          
        
          
        
          
          <a class="page-link" href="/software/">Software</a>
          
        
          
        
          
        
          
        
	<!-- special link for tutorials pages, which are done with MkDocs -->
        <a class="page-link" href="/tutorials">Tutorials</a>
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <head>
  <link rel="shortcut icon" href="/mauve/favicon.ico" />
</head>




  <table>
  <tr>
  <td width="175" valign="top">
  
<h4 id="download-mauve"><a href="/mauve/download.html"><strong>Download Mauve</strong></a></h4>

<h4 id="user-guide"><strong>User Guide:</strong></h4>
<ul>
  <li><a href="/mauve/user-guide/introduction.html">Introduction</a></li>
  <li><a href="/mauve/user-guide/installing.html">Installing</a></li>
  <li><a href="/mauve/user-guide/aligning.html">Aligning genomes</a></li>
  <li><a href="/mauve/user-guide/viewer.html">The viewer</a></li>
  <li><a href="/mauve/user-guide/files.html">File formats</a></li>
  <li><a href="/mauve/user-guide/reordering.html">Reordering contigs</a></li>
  <li><a href="/mauve/user-guide/mauvealigner.html">mauveAligner</a></li>
  <li><a href="/mauve/user-guide/progressivemauve.html">progressiveMauve</a></li>
  <li><a href="/mauve/user-guide/versions.html">Version history</a></li>
  <li><a href="/mauve/user-guide/screenshots.html">Screenshots</a></li>
</ul>

<h4 id="developer-guide"><strong>Developer Guide:</strong></h4>
<ul>
  <li><a href="/mauve/developer-guide/overview.html">Overview</a></li>
  <li><a href="/mauve/developer-guide/building.html">Building from source</a></li>
  <li><a href="/mauve/developer-guide/developing-the-gui.html">Developing the GUI</a></li>
  <li><a href="/mauve/developer-guide/building-on-windows.html">Windows builds</a></li>
  <li><a href="/mauve/developer-guide/building-on-mac.html">Mac builds</a></li>
  <li><a href="/mauve/developer-guide/deploying-mauve-releases.html">Deploying Mauve</a></li>
  <li><a href="/mauve/developer-guide/benchmarking.html">Benchmarking</a></li>
</ul>


  </td>
  <td valign="top">  


<div class="post">

  <article class="post-content">
    <h3 id="mauvealigner-command-line-usage">mauveAligner command-line usage</h3>

<p>The mauveAligner and progressiveMauve alignment algorithms have been implemented as command-line programs included with the downloadable Mauve software.  When run from the command-line, these programs provide options not yet available in the graphical interface.</p>

<h3 id="where-to-find-mauvealigner-and-progressivemauve">Where to find mauveAligner and progressiveMauve</h3>

<p>mauveAligner and progressiveMauve are included with the Mauve software. On Windows they are located in the directory where Mauve was installed, usually <code class="language-plaintext highlighter-rouge">C:\Program Files\Mauve\</code>.  On 64-bit Windows platforms, the binaries in the win64 subdirectory may be used instead.  On Mac OS X the mauveAligner and progressiveMauve binaries are packaged within the Mauve application. Relative to Mauve.app the paths will be <code class="language-plaintext highlighter-rouge">Mauve.app/Contents/MacOS/mauveAligner</code> and <code class="language-plaintext highlighter-rouge">Mauve.app/Contents/MacOS/progressiveMauve</code>. The binaries can be copied to a more convenient location. However, if using Mauve versions 2.0.0 or earlier, be sure to also copy the muscle_aed binary to the same location. On Linux the mauveAligner and progressiveMauve binaries are in the top level directory of the software distribution, with 64-bit binaries in the linux-x64 subdirectory.</p>

<p>Alternatively, mauveAligner and progressiveMauve can be compiled from source for your particular platform. See the <a href="/mauve/developer-guide/overview.html">Mauve Developer’s Guide</a> for more details.</p>

<h3 id="mauvealigner-arguments-and-examples">mauveAligner arguments and examples</h3>

<p>Usage:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner [options] &lt;seq1 filename&gt; &lt;sml1 filename&gt; ... &lt;seqN filename&gt; &lt;smlN filename&gt;</code></p>

<p>or</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner [options] &lt;Multi-FastA file&gt;</code></p>

<p>Each entry like <code class="language-plaintext highlighter-rouge">&lt;seq filename&gt;</code> denotes the name of a sequence file on disk that contains a genome to align. Sequence files can be in FastA, Multi-FastA, or GenBank flat file format. If an individual file contains several sequence entries, they will be concatenated and the whole concatenated sequence will be aligned to the sequences in the other files.</p>

<p>Each sequence must have a corresponding Sorted Mer List (SML) file name given. If the SML file does not exist, mauveAligner will create it. Alternatively, mauveAligner can be given the name of a Multi-FastA file containing all the genomes to align. In this case mauveAligner assumes one genome per Multi-FastA sequence entry and will align the entries to each other.</p>

<p>Example 1. Aligning <em>E. coli</em> and <em>Salmonella</em> genomes stored in separate GenBank files:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner --output=ec_sa.mauve --output-alignment=ec_vs_sa.alignment ecoli.gbk ecoli.gbk.sml salmonella.gbk salmonella.gbk.sml</code></p>

<p>Example 2. Aligning several poxvirus genomes stored in a single Multi-FastA file:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner --output=poxvirus.mauve --output-alignment=poxvirus.alignment all_poxvirus.mfa</code></p>

<p>Example 3. Generating initial alignment anchors without actually aligning:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner --no-recursion --no-gapped-alignment -o poxvirus.mauve all_poxvirus.mfa</code></p>

<p>Example 4. Aligning a group of Hepatitis C virii known to have no rearrangements:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner --collinear --output=hcv.mauve --output-alignment=hcv.alignment hcv.mfa</code></p>

<p>Example 5. Aligning two yeast genomes, increasing the weight threshold to avoid spurious LCBs:</p>

<p><code class="language-plaintext highlighter-rouge">mauveAligner --weight=300 --output=two_yeast.mauve --output-alignment=two_yeast.alignment S_cerevisiae.gbk S_cerevisiae.gbk.sml S_bayanus.gbk S_bayanus.gbk.sml</code></p>

<p>Many other command line options can be given to the mauveAligner binary. A full listing follows.</p>

<h4 id="general-options">General options:</h4>

<p><code class="language-plaintext highlighter-rouge">--output=&lt;file&gt;</code> Output file name. Prints to screen by default</p>

<p><code class="language-plaintext highlighter-rouge">--mums</code> Find MUMs only, do not attempt to determine locally collinear blocks (LCBs)</p>

<p><code class="language-plaintext highlighter-rouge">--no-recursion</code> Don’t perform recursive anchor identification (implies –no-gapped-alignment)</p>

<p><code class="language-plaintext highlighter-rouge">--no-lcb-extension</code> If determining LCBs, don’t attempt to extend the LCBs</p>

<p><code class="language-plaintext highlighter-rouge">--seed-size=&lt;number&gt;</code> Initial seed match size, default is log2( average seq. length )</p>

<p><code class="language-plaintext highlighter-rouge">--max-extension-iterations=&lt;number&gt;</code> Limit LCB extensions to this number of attempts, default is 4</p>

<p><code class="language-plaintext highlighter-rouge">--eliminate-inclusions</code> Eliminate linked inclusions in subset matches.</p>

<p><code class="language-plaintext highlighter-rouge">--weight=&lt;number&gt;</code> Minimum LCB weight in base pairs per sequence</p>

<p><code class="language-plaintext highlighter-rouge">--match-input=&lt;file&gt;</code> Use specified match file instead of searching for matches</p>

<p><code class="language-plaintext highlighter-rouge">--lcb-match-input</code> Indicates that the match input file contains matches that have been clustered into LCBs</p>

<p><code class="language-plaintext highlighter-rouge">--lcb-input=&lt;file&gt;</code> Use specified lcb file instead of constructing LCBs (skips LCB generation)</p>

<p><code class="language-plaintext highlighter-rouge">--scratch-path=&lt;path&gt;</code> For large genomes, use a directory for storage of temporary data. Should be given two or more times to with different paths.</p>

<p><code class="language-plaintext highlighter-rouge">--id-matrix=&lt;file&gt;</code> Generate LCB stats and write them to the specified file</p>

<p><code class="language-plaintext highlighter-rouge">--island-size=&lt;number&gt;</code> Find islands larger than the given number</p>

<p><code class="language-plaintext highlighter-rouge">--island-output=&lt;file&gt;</code> Output islands the given file (requires –island-size)</p>

<p><code class="language-plaintext highlighter-rouge">--backbone-size=&lt;number&gt;</code> Find stretches of backbone longer than the given number of b.p.</p>

<p><code class="language-plaintext highlighter-rouge">--max-backbone-gap=&lt;number&gt;</code> Allow backbone to be interrupted by gaps up to this length in b.p.</p>

<p><code class="language-plaintext highlighter-rouge">--backbone-output=&lt;file&gt;</code> Output islands the given file (requires –island-size)</p>

<p><code class="language-plaintext highlighter-rouge">--coverage-output=&lt;file&gt;</code> Output a coverage list to the specified file (- for stdout)</p>

<p><code class="language-plaintext highlighter-rouge">--repeats</code> Generates a repeat map. Only one sequence can be specified</p>

<p><code class="language-plaintext highlighter-rouge">--output-guide-tree=&lt;file&gt;</code> Write out the guide tree used for CLUSTALW alignment to the designated file</p>

<p><code class="language-plaintext highlighter-rouge">--collinear</code> Assume that input sequences are collinear–they have no rearrangements</p>

<h4 id="gapped-alignment-controls">Gapped alignment controls:</h4>

<p><code class="language-plaintext highlighter-rouge">--no-gapped-alignment</code> Don’t perform a gapped alignment</p>

<p><code class="language-plaintext highlighter-rouge">--gapped-aligner=&lt;muscle|clustal&gt;</code> Set the gapped alignment algorithm (Default is muscle)</p>

<p><code class="language-plaintext highlighter-rouge">--max-gapped-aligner-length=&lt;number&gt;</code> Maximum number of base pairs to attempt aligning with the gapped aligner</p>

<p><code class="language-plaintext highlighter-rouge">--min-recursive-gap-length=&lt;number&gt;</code> Minimum size of gaps that Mauve will perform recursive MUM anchoring on (Default is 200)</p>

<h4 id="signed-permutation-matrix-options">Signed permutation matrix options:</h4>

<p><code class="language-plaintext highlighter-rouge">--permutation-matrix-output=&lt;file&gt;</code> Write out the LCBs as a signed permutation matrix to the given file</p>

<p><code class="language-plaintext highlighter-rouge">--permutation-matrix-min-weight=&lt;number&gt;</code> A permutation matrix will be written for every set of LCBs with weight between this value and the value of –weight</p>

<h4 id="alignment-output-options">Alignment output options:</h4>

<p><code class="language-plaintext highlighter-rouge">--alignment-output-dir=&lt;directory&gt;</code> Outputs a set of alignment files (one per LCB) to a given directory</p>

<p><code class="language-plaintext highlighter-rouge">--alignment-output-format=&lt;directory&gt;</code> Selects the output format for –alignment-output-dir</p>

<p><code class="language-plaintext highlighter-rouge">--output-alignment=&lt;file&gt;</code> Write out an XMFA format alignment to the designated file</p>

<h3 id="mauvealigner-execution-modes">mauveAligner Execution modes</h3>

<h4 id="mum-match-generation">MUM (Match) generation</h4>

<p>Generates a list of unique and ungapped local multiple alignments shared by two or more of the sequences. Alignments among subsets of the input genomes are included unless the <code class="language-plaintext highlighter-rouge">--nway-filter</code> option is used. See the MUM file format reference.</p>

<p>Use the <code class="language-plaintext highlighter-rouge">--mums</code> command line option to invoke match generation without alignment. Other command line options that can be used in combination with <code class="language-plaintext highlighter-rouge">--mums</code> are <code class="language-plaintext highlighter-rouge">--seed-size</code> to specify a minimum weight for matches, and –eliminate-inclusions to remove overlapping regions among different matches. Acceptable values for seed size are odd numbers in the range of 5 to 31. The algorithm to remove overlapping regions gives preference to matches with higher multiplicity, where multiplicity is defined as the number of genomes that a match occurs in. When two overlapping matches have identical multiplicity, the shorter of the two matches is truncated to eliminate the overlap.</p>

<h4 id="breakpoint-elimination">Breakpoint elimination</h4>

<p>Performs greedy breakpoint elimination to determine Locally Collinear Blocks (LCBs). The resulting LCBs will all be above some minimum weight.  Invoked when full alignment is disabled by adding –no-gapped-alignment.</p>

<h4 id="full-alignment">Full alignment</h4>

<p>Performs a complete alignment of the conserved regions among the input genome sequences using ClustalW progressive dynamic programming.  This is the default execution mode.</p>

  </article>

</div>


  </td>
  </table>



      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li>the Darling lab | computational (meta)genomics</li>
          <li><a href="mailto:aaron.darling@uts.edu.au">aaron.darling@uts.edu.au</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/koadman">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          

          
          <li>
            <a href="https://twitter.com/koadman">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
</p>
      </div>
    </div>
    <img alt="tree of bacterial and archaeal life, showing human gut lineages in blue" src="/assets/tree.png">

  </div>
</footer>


  </body>

</html>
