<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width initial-scale=1">

  <title>the Darling lab | computational (meta)genomics</title>
  <meta name="description" content="The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="https://darlinglab.org/mauve/user-guide/progressivemauve.html">
  <link rel="alternate" type="application/atom+xml" title="the Darling lab | computational (meta)genomics" href="https://darlinglab.org/feed.xml" />
  <link rel="shortcut icon" href="/site32.jpg" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">the Darling lab | computational (meta)genomics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/mauve/mauve.html">Mauve</a>
          
        
          
        
          
          <a class="page-link" href="/openings/">Openings</a>
          
        
          
        
          
          <a class="page-link" href="/people/">People</a>
          
        
          
        
          
          <a class="page-link" href="/projects/">Projects</a>
          
        
          
          <a class="page-link" href="/publications/">Publications</a>
          
        
          
        
          
          <a class="page-link" href="/reviews/">Reviews</a>
          
        
          
        
          
          <a class="page-link" href="/software/">Software</a>
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <head>
  <link rel="shortcut icon" href="/mauve/favicon.ico" />
</head>




  <table>
  <tr>
  <td width="175" valign="top">
  
<h4 id="download-mauve"><a href="/mauve/download.html"><strong>Download Mauve</strong></a></h4>

<h4 id="user-guide"><strong>User Guide:</strong></h4>
<ul>
  <li><a href="/mauve/user-guide/introduction.html">Introduction</a></li>
  <li><a href="/mauve/user-guide/installing.html">Installing</a></li>
  <li><a href="/mauve/user-guide/aligning.html">Aligning genomes</a></li>
  <li><a href="/mauve/user-guide/viewer.html">The viewer</a></li>
  <li><a href="/mauve/user-guide/files.html">File formats</a></li>
  <li><a href="/mauve/user-guide/reordering.html">Reordering contigs</a></li>
  <li><a href="/mauve/user-guide/mauvealigner.html">mauveAligner</a></li>
  <li><a href="/mauve/user-guide/progressivemauve.html">progressiveMauve</a></li>
  <li><a href="/mauve/user-guide/versions.html">Version history</a></li>
  <li><a href="/mauve/user-guide/screenshots.html">Screenshots</a></li>
</ul>

<h4 id="developer-guide"><strong>Developer Guide:</strong></h4>
<ul>
  <li><a href="/mauve/developer-guide/overview.html">Overview</a></li>
  <li><a href="/mauve/developer-guide/building.html">Building from source</a></li>
  <li><a href="/mauve/developer-guide/developing-the-gui.html">Developing the GUI</a></li>
  <li><a href="/mauve/developer-guide/building-on-windows.html">Windows builds</a></li>
  <li><a href="/mauve/developer-guide/building-on-mac.html">Mac builds</a></li>
  <li><a href="/mauve/developer-guide/deploying-mauve-releases.html">Deploying Mauve</a></li>
  <li><a href="/mauve/developer-guide/benchmarking.html">Benchmarking</a></li>
</ul>


  </td>
  <td valign="top">  


<div class="post">

  <article class="post-content">
    <h3 id="using-progressivemauve-from-the-command-line">Using progressiveMauve from the command-line</h3>

<p>progressiveMauve also offers many command-line options to provide detailed control over alignment parameters.  This section gives a reference of the parameters and some examples regarding their usage.</p>

<p>Before running any of the example commands it will be necessary to locate the progressiveMauve binary on your system, as described in the <a href="/mauve/user-guide/mauvealigner.html">mauveAligner command-line page</a>.</p>

<h4 id="command-line-paramter-reference">Command-line paramter reference</h4>

<p><code class="language-plaintext highlighter-rouge">--apply-backbone=&lt;file&gt;</code> Read an existing sequence alignment in XMFA format and apply backbone statistics to it</p>

<p><code class="language-plaintext highlighter-rouge">--disable-backbone</code> Disable backbone detection</p>

<p><code class="language-plaintext highlighter-rouge">--mums</code> Find MUMs only, do not attempt to determine locally collinear blocks (LCBs)</p>

<p><code class="language-plaintext highlighter-rouge">--seed-weight=&lt;number&gt;</code> Use the specified seed weight for calculating initial anchors</p>

<p><code class="language-plaintext highlighter-rouge">--output=&lt;file&gt;</code> Output file name. Prints to screen by default</p>

<p><code class="language-plaintext highlighter-rouge">--backbone-output=&lt;file&gt;</code> Backbone output file name (optional).</p>

<p><code class="language-plaintext highlighter-rouge">--match-input=&lt;file&gt;</code> Use specified match file instead of searching for matches</p>

<p><code class="language-plaintext highlighter-rouge">--input-id-matrix=&lt;file&gt;</code> An identity matrix describing similarity among all pairs of input sequences/alignments</p>

<p><code class="language-plaintext highlighter-rouge">--max-gapped-aligner-length=&lt;number&gt;</code> Maximum number of base pairs to attempt aligning with the gapped aligner</p>

<p><code class="language-plaintext highlighter-rouge">--input-guide-tree=&lt;file&gt;</code> A phylogenetic guide tree in NEWICK format that describes the order in which sequences will be aligned</p>

<p><code class="language-plaintext highlighter-rouge">--output-guide-tree=&lt;file&gt;</code> Write out the guide tree used for alignment to a file</p>

<p><code class="language-plaintext highlighter-rouge">--version</code> Display software version information</p>

<p><code class="language-plaintext highlighter-rouge">--debug</code> Run in debug mode (perform internal consistency checks–very slow)</p>

<p><code class="language-plaintext highlighter-rouge">--scratch-path-1=&lt;path&gt;</code> Designate a path that can be used for temporary data storage. Two or more paths should be specified.</p>

<p><code class="language-plaintext highlighter-rouge">--scratch-path-2=&lt;path&gt;</code> Designate a path that can be used for temporary data storage. Two or more paths should be specified.</p>

<p><code class="language-plaintext highlighter-rouge">--collinear</code> Assume that input sequences are collinear–they have no rearrangements</p>

<p><code class="language-plaintext highlighter-rouge">--scoring-scheme=&lt;ancestral|sp_ancestral|sp&gt;</code> Selects the anchoring score function. Default is extant sum-of-pairs (sp).</p>

<p><code class="language-plaintext highlighter-rouge">--no-weight-scaling</code> Don’t scale LCB weights by conservation distance and breakpoint distance</p>

<p><code class="language-plaintext highlighter-rouge">--max-breakpoint-distance-scale=&lt;number [0,1]&gt;</code> Set the maximum weight scaling by breakpoint distance. Defaults to 0.9</p>

<p><code class="language-plaintext highlighter-rouge">--conservation-distance-scale=&lt;number [0,1]&gt;</code> Scale conservation distances by this amount. Defaults to 1</p>

<p><code class="language-plaintext highlighter-rouge">--skip-refinement</code> Do not perform iterative refinement</p>

<p><code class="language-plaintext highlighter-rouge">--skip-gapped-alignment</code> Do not perform gapped alignment</p>

<p><code class="language-plaintext highlighter-rouge">--bp-dist-estimate-min-score=&lt;number&gt;</code> Minimum LCB score for estimating pairwise breakpoint distance</p>

<p><code class="language-plaintext highlighter-rouge">--mem-clean</code> Set this to true when debugging memory allocations</p>

<p><code class="language-plaintext highlighter-rouge">--gap-open=&lt;number&gt;</code> Gap open penalty</p>

<p><code class="language-plaintext highlighter-rouge">--gap-extend=&lt;number&gt;</code> Gap extend penalty</p>

<p><code class="language-plaintext highlighter-rouge">--substitution-matrix=&lt;file&gt;</code> Nucleotide substitution matrix in NCBI format</p>

<p><code class="language-plaintext highlighter-rouge">--weight=&lt;number&gt;</code> Minimum pairwise LCB score</p>

<p><code class="language-plaintext highlighter-rouge">--min-scaled-penalty=&lt;number&gt;</code> Minimum breakpoint penalty after scaling the penalty by expected divergence</p>

<p><code class="language-plaintext highlighter-rouge">--hmm-p-go-homologous=&lt;number&gt;</code> Probability of transitioning from the unrelated to the homologous state [0.0001]</p>

<p><code class="language-plaintext highlighter-rouge">--hmm-p-go-unrelated=&lt;number&gt;</code> Probability of transitioning from the homologous to the unrelated state [0.000001]</p>

<p><code class="language-plaintext highlighter-rouge">--seed-family</code> Use a family of spaced seeds to improve sensitivity</p>

<h4 id="examples-of-progressivemauve-usage">Examples of progressiveMauve usage</h4>

<p><strong>Example 1.</strong>  Align three genomes from the input files genome_1.gbk, genome_2.gbk, and genome_3.gbk, saving the output to a file called threeway.xmfa</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --output=threeway.xmfa genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 2.</strong>  Align the same three genomes but also save the guide tree and produce a backbone file</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --output=threeway.xmfa --output-guide-tree=threeway.tree --backbone-output=threeway.backbone genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 3.</strong>  Align the same three genomes, but do not detect forced alignment of unrelated sequence and do not create a backbone file</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --output=threeway_no_backbone.xmfa --disable-backbone genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 4.</strong>  Detect forced alignment of unrelated sequence in the alignment produced in Example 3.  Use custom Homology HMM transition parameters.  Save a backbone file.</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --apply-backbone=threeway_no_backbone.xmfa --output=threeway.xmfa --backbone-output=threeway.backbone --hmm-p-go-homologous=0.001 --hmm-p-go-unrelated=0.000005</code></p>

<p><strong>Example 5.</strong>  Compute ungapped local-multiple alignments among the input sequences and save them to a file called threeway.mums</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --mums --output=threeway.mums genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 6.</strong>  Compute an alignment of the same three genomes, using previously computed local-multiple alignments</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --match-input=threeway.mums --output=threeway.xmfa genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 7.</strong>  Set a custom breakpoint penalty to cope with genomes where default penalty does not work.  The default penalty can be extracted from the program’s textual output, in this hypothetical example, the default penalty will be 100000.</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --output=threeway.xmfa --weight=50000 genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 8.</strong>  Set a minimum scaled breakpoint penalty to cope with the case where most genomes are aligned correctly, but manual inspection reveals that a divergent genome has too many predicted rearrangements.</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --min-scaled-penalty=5000 --output=threeway.xmfa  genome_1.gbk genome_2.gbk genome_3.gbk</code></p>

<p><strong>Example 9.</strong>  Globally align a set of collinear virus genomes that reside in a single FastA file.  Use seed families to improve anchoring sensitivity in regions below 70% sequence identity.</p>

<p><code class="language-plaintext highlighter-rouge">progressiveMauve --collinear --seed-family --disable-backbone --output=virus.xmfa all_virii.fasta</code></p>

  </article>

</div>


  </td>
  </table>



      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li>the Darling lab | computational (meta)genomics</li>
          <li><a href="mailto:aaron.darling@uts.edu.au">aaron.darling@uts.edu.au</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/koadman">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          

          
          <li>
            <a href="https://twitter.com/koadman">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
</p>
      </div>
    </div>
    <img alt="tree of bacterial and archaeal life, showing human gut lineages in blue" src="/assets/tree.png">

  </div>
</footer>


  </body>

</html>
