<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width initial-scale=1">

  <title>the Darling lab | computational (meta)genomics</title>
  <meta name="description" content="The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
">

  <link rel="stylesheet" href="/platypusfarm//css/main.css">
  <link rel="canonical" href="https://koadman.github.io/platypusfarm//mauve/user-guide/versions.html">
  <link rel="alternate" type="application/atom+xml" title="the Darling lab | computational (meta)genomics" href="https://koadman.github.io/platypusfarm//feed.xml" />
  <link rel="shortcut icon" href="/platypusfarm//site32.jpg" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/platypusfarm//">the Darling lab | computational (meta)genomics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//mauve/mauve.html">Mauve</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//openings/">Openings</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//people/">People</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//projects/">Projects</a>
          
        
          
          <a class="page-link" href="/platypusfarm//publications/">Publications</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//reviews/">Reviews</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm//software/">Software</a>
          
        
          
        
          
        
          
        
	<!-- special link for tutorials pages, which are done with MkDocs -->
        <a class="page-link" href="/platypusfarm//tutorials">Tutorials</a>
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <head>
  <link rel="shortcut icon" href="/platypusfarm//mauve/favicon.ico" />
</head>




  <table>
  <tr>
  <td width="175" valign="top">
  
<h4 id="download-mauve"><a href="/platypusfarm//mauve/download.html"><strong>Download Mauve</strong></a></h4>

<h4 id="user-guide"><strong>User Guide:</strong></h4>
<ul>
  <li><a href="/platypusfarm//mauve/user-guide/introduction.html">Introduction</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/installing.html">Installing</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/aligning.html">Aligning genomes</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/viewer.html">The viewer</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/files.html">File formats</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/reordering.html">Reordering contigs</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/mauvealigner.html">mauveAligner</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/progressivemauve.html">progressiveMauve</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/versions.html">Version history</a></li>
  <li><a href="/platypusfarm//mauve/user-guide/screenshots.html">Screenshots</a></li>
</ul>

<h4 id="developer-guide"><strong>Developer Guide:</strong></h4>
<ul>
  <li><a href="/platypusfarm//mauve/developer-guide/overview.html">Overview</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/building.html">Building from source</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/developing-the-gui.html">Developing the GUI</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/building-on-windows.html">Windows builds</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/building-on-mac.html">Mac builds</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/deploying-mauve-releases.html">Deploying Mauve</a></li>
  <li><a href="/platypusfarm//mauve/developer-guide/benchmarking.html">Benchmarking</a></li>
</ul>


  </td>
  <td valign="top">  


<div class="post">

  <article class="post-content">
    <h3 id="mauve-version-history">Mauve version history</h3>

<h4 id="version-240-2014-12-21">Version 2.4.0 (2014-12-21)</h4>

<h5 id="new-features">New features:</h5>
<ul>
  <li>Compatibility with Mac OS X 10.8+</li>
  <li>Mauve Assembly Metrics <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/pmid/21810901/">(publication)</a></li>
  <li>SNP output from the command line</li>
  <li>Permutation matrix output</li>
  <li>Updated visual styling of similarity plots</li>
  <li>Mauve Contig Mover can now run headless (command-line) with OpenJDK and Oracle JDK 7 and 8</li>
</ul>

<h5 id="bugs-fixed">Bugs fixed:</h5>
<ul>
  <li>Poor performance of annotation tooltips on mouseover</li>
  <li>Errors reading GenBank annotation from RAST and Prokka (thanks Torsten Seemann &amp; Matt DeMaere)</li>
  <li>Alignment of genomes &gt; 2Gbp</li>
  <li>new OpenJDK fonts caused ugly GUI component layouts</li>
  <li>Build incompatibilities with modern Boost libs</li>
  <li>Contig N50 off-by-one error in Mauve Assembly Metrics</li>
  <li>Boundary condition bugs in Mauve Assembly Metrics</li>
</ul>

<h5 id="known-bugs">Known bugs:</h5>
<ul>
  <li>The OS X build uses an older version of the aligner component</li>
  <li>The aligner is not supported on Windows 8. Alignments can be viewed but not created</li>
</ul>

<h4 id="version-231-2009-11-11">Version 2.3.1 (2009-11-11)</h4>

<h5 id="new-features-1">New features:</h5>
<ul>
  <li>Compatibility with older versions of Mac OS X</li>
</ul>

<h5 id="bugs-fixed-1">Bugs fixed:</h5>
<ul>
  <li>SNP output now handles mixed case sequence files and gap characters</li>
  <li>Aligner memory usage has been streamlined, resulting in reduced memory requirements for alignments with many genomes</li>
  <li>Fix for bug that prevented reordering GenBank-format draft genomes on the command-line</li>
  <li>Fixed a crash when generating mums</li>
  <li>Report an error when sequences containing gaps are used as input</li>
  <li>Remove extra newlines in the ortholog alignment output</li>
  <li>Fixed an OpenJDK rectangle drawing bug</li>
</ul>

<h4 id="version-230-2009-06-16">Version 2.3.0 (2009-06-16)</h4>

<h5 id="major-new-features">Major new features:</h5>
<ul>
  <li>Reordering contigs of draft genomes, published as Rissman et al 2009</li>
  <li>Export a list of Single Nucleotide Polymorphisms present in an alignment (thanks to Meg Woolfit for the suggestion)</li>
  <li>Export a file listing the genome arrangement as a signed permutation</li>
  <li>Export a list of positionally orthologous CDS, tRNAs, or misc_RNAs (thanks to Elizabeth Skippington for the suggestion)</li>
  <li>Export a file containing alignments of orthologous CDS, including alignments of unannotated regions predicted to be orthologous (thanks to Sam Sheppard for the idea)</li>
</ul>

<h5 id="other-new-features">Other new features:</h5>
<ul>
  <li>On Linux it is now possible to launch Mauve from outside the Mauve directory (thx Dongying Wu)</li>
  <li>Option to disable drawing the mouse cursor highlighting orthologous regions so that figures for manuscripts can be rendered (thanks to Andrew Kropinski for the idea)</li>
</ul>

<h5 id="bugs-fixed-2">Bugs fixed:</h5>
<ul>
  <li>An alignment would complete successfully but the viewer would fail to load it because the alignment contained invalid data in the backbone file.  This bug was very commonly encountered in 2.2.0.</li>
  <li>An alignment with progressiveMauve would crash with an error like “cga doesn’t fit” or another error message, when run on multicore computers.  This was a race condition in progressiveMauve’s parallelism.
    <h5 id="known-issues">Known issues:</h5>
  </li>
  <li>Parallelism has been disabled for all supported platforms to avoid a race condition bug.</li>
</ul>

<h4 id="version-220-2008-06-23">Version 2.2.0 (2008-06-23)</h4>

<h5 id="new-features-2">New features:</h5>
<ul>
  <li>New menu option to disable the red contig boundary lines</li>
  <li>Progressive aligner sensitivity/accuracy improvements</li>
</ul>

<h5 id="bugs-fixed-3">Bugs fixed:</h5>
<ul>
  <li>progressiveMauve was not sensitive enough to small rearrangements, although it did have high predictive value</li>
  <li>progressiveMauve was not properly extending LCBs in the 2.1.x releases</li>
  <li>progressiveMauve did not work on 64-bit linux (indefinite waiting message)</li>
  <li>Versions 2.1.x did not render vector graphics when printing to PDF or EPS</li>
  <li>seed families are too memory intensive and are now disabled by default</li>
  <li>segmentation fault (crash) on linux/mac due to a bug in c++ sort functions</li>
  <li>update check causes a stall during GUI launch when the network is inaccessible</li>
  <li>building from source was difficult</li>
</ul>

<h5 id="known-issues-1">Known issues:</h5>
<ul>
  <li>Mauve 2.2.0 for Macintosh can not run in parallel on more than one CPU</li>
  <li>Additional bugs can be found at http://sourceforge.net/tracker/?group_id=181544&amp;atid=897627</li>
</ul>

<h4 id="version-211-2007-11-27">Version 2.1.1 (2007-11-27)</h4>

<h5 id="bugs-fixed-4">Bugs fixed:</h5>
<ul>
  <li>The alignment programs mauveAligner and progressiveMauve did not work on Mac due to a missing shared library (libgomp.dylib)</li>
  <li>progressiveMauve would occasionally crash during alignment scoring and greedy breakpoint elimination due to a math bug</li>
  <li>The sequence navigator did not properly highlight feature search results when launched from the menu.  It did work when launched from the toolbar binoculars button.</li>
</ul>

<h4 id="version-210-2007-11-20">Version 2.1.0 (2007-11-20)</h4>

<h5 id="new-features-3">New features:</h5>
<ul>
  <li>A graphical interface to reorder and hide genomes from the display, and to change the reference genome</li>
  <li>A new graphical interface to change the viewing style for genome comparisons</li>
  <li>Parallelized alignment algorithms that run faster on multi-core CPUs like the Intel Core Duo and the AMD Opteron</li>
  <li>64-bit implementations for Windows and Mac OS X.  64-bit machines with enough RAM can align 20 or more bacterial genomes</li>
  <li>Seed family searches.  Several seed patterns can be used simultaneously to improve sensitivity on heavily diverged genomes</li>
</ul>

<h5 id="bugs-fixed-5">Bugs fixed:</h5>
<ul>
  <li>Several progressiveMauve alignment crashes</li>
  <li>A mauveAligner bug whereby the aligner never finished on some datasets</li>
  <li>Many more.  For a full list of known and fixed bugs please see http://sourceforge.net/tracker/?group_id=181544&amp;atid=897627</li>
</ul>

<h5 id="known-issues-2">Known issues:</h5>
<ul>
  <li>Mauve 2.1.0 for Macintosh requires OS X version 10.4 or later</li>
  <li>A bug exists which can crash Mauve on Linux and Mac OS X during the “recursive anchor search.”</li>
  <li>Mauve does not work with GNU java, which is the default java in some Linux distributions</li>
  <li>Sun JRE 1.6.0 for Linux has a bug that breaks alignment loading (it’s not my fault!).  It may be necessary to manually open the alignment file using “File-&gt;Open” after making the alignment.</li>
</ul>

<h4 id="version-200-2007-01-26">Version 2.0.0 (2007-01-26)</h4>
<p>Mauve 2.0 includes an accurate new alignment algorithm called Progressive Mauve.  The progressive alignment algorithm has been described in Aaron Darling’s Ph.D. thesis.</p>

<h5 id="progressive-mauve-features">Progressive Mauve features:</h5>
<ul>
  <li>Substantially improved sensitivity on divergent genomes.  Progressive Mauve can “climb a phylogenetic ladder” such that a pair of genomes with &gt; 1 substitution per site can be aligned if phylogenetic intermediate genomes are included in the alignment</li>
  <li>Segments that are present in subsets of the genomes will be aligned.  The original Mauve aligned only segments conserved in all genomes.</li>
  <li>A configurable substitution matrix dictates the LCB and alignment scores.  The substitution matrix gives Mauve additional power in distinguishing true homology from random sequence similarity.  In the original algorithm LCBs were scored based on the sum of multi-MUM lengths</li>
  <li>A random-walk homology test  identifies conserved “backbone” segments.  Random walk statistics similar to those used in Gapped-BLAST are used to identify pairwise conserved segments.  Pairwise homologies are then combined by transitive homology relationships to identify segments conserved in two or more genomes.</li>
  <li>progressiveMauve can scale to hundreds of (small) genomes.  We have successfully aligned over a hundred poxvirus genomes, hundreds of Hepatitis C Virus genomes, and the genomes of over thirty enterobacteria.</li>
  <li>
    <p>Anna Rissman has contributed a new annotation search interface that tremendously improves browsing genomes with annotated sequence features.  The new search interface can:</p>
  </li>
  <li>Jump directly to a particular numerical coordinate in a particular genome</li>
  <li>Jump directly to an annotated feature of a particular name</li>
  <li>Search for and list annotated features by any part of their name or qualifying value.  Searchable fields include things like CDS name, product description, GO terms, database x-refs, and amino acid subsequences</li>
</ul>

<h5 id="other-new-features-1">Other new features:</h5>
<ul>
  <li>The windows installer supports installations by users without administrative privileges and also supports multiple concurrent installations</li>
  <li>The windows installer includes Java 6 for machines without an existing Java installation</li>
  <li>Alignments from progressiveMauve can be viewed in “backbone color” mode, which colorizes segments based on the combination of genomes in which they are conserved.</li>
</ul>

<h5 id="bugs-fixed-6">Bugs fixed:</h5>
<ul>
  <li>Infinite looping during LCB extension should be fixed.  LCB extension may still be slow for some datasets, with questionable benefit.  It can be disabled in the Parameters tab of the “Align sequences..” window.</li>
  <li>Several bugs were fixed that would cause sporadic crashes in MUSCLE 3.6.</li>
</ul>

<h5 id="known-bugs-1">Known bugs:</h5>
<ul>
  <li>Bug tracking has moved to the sourceforge web site.  Pease use the following address to report bugs:  http://sourceforge.net/tracker/?group_id=181544&amp;atid=897627</li>
</ul>

<h4 id="version-130-2006-05-24">Version 1.3.0 (2006-05-24)</h4>

<h5 id="new-features-4">New features:</h5>
<ul>
  <li>Rearrangement history analysis by generalized transposition/block interchange, code courtesy of Mike Tsai. Use the green/blue/black toolbar button to perform an analysis. For details on the inference algorithm see Yancopoulos, S., Attie, O and Friedberg, R. “Efficient sorting of genomic permutations by translocation, inversion and block interchange” Bioinformatics 2005 21(16):3340-3346</li>
  <li>Rearrangement history through the GRIMM/MGR server now supported, use the red/yellow/black toolbar button.</li>
  <li>No more limit on the number of input sequences. This feature comes at the cost of increased memory consumption. Just because the limit has been removed doesn’t mean that the alignment algorithm will do a good job handling a large number of sequences.</li>
  <li>The mauveAligner command-line tool has the new option –lcb-match-input to extract matches from an existing alignment and use them as anchors when computing a new alignment.</li>
  <li>MUSCLE has been upgraded to version 3.6.</li>
  <li>The Mac OS X version is now a Universal binary with support for both Intel and PowerPC macs.</li>
</ul>

<h5 id="bugs-fixed-7">Bugs fixed:</h5>
<ul>
  <li>MUSCLE on Windows now runs faster and uses less memory.</li>
  <li>On unix, Mauve will search $PATH for mauveAligner.</li>
  <li>Eliminated the obnoxious behavior where Mauve would repeatedly pop up the console window during the alignment process</li>
  <li>Fixed bugs when viewing alignments after shifting the LCB minimum weight slider.</li>
</ul>

<h4 id="version-123-2005-10-20">Version 1.2.3 (2005-10-20)</h4>

<h5 id="new-features-5">New features:</h5>
<ul>
  <li>Display each component of multi-part features (e.g. a multiple exon gene) as a linked series of boxes</li>
  <li>Display repeat_region annotations from GenBank files</li>
  <li>Improved load time for previously viewed alignments using a disk-based alignment cache</li>
  <li>Added a help menu item that will clear the on-disk alignment cache</li>
  <li>Colorized the similarity plots, grey was starting to depress our entire lab.</li>
  <li>Support for highlighting arbitrary regions of the sequence by holding the shift key while clicking and dragging on the display</li>
  <li>Faster nucleotide display when zoomed in closely</li>
</ul>

<h5 id="bug-fixes">Bug fixes:</h5>
<ul>
  <li>Fixed GUI interface for setting the minimum seed size (Thanks to Bill Bruno and Todd Treangen)</li>
  <li>Fixed mouseover, highlighting, and display alignment in the LCB display mode (used when the Full Alignment option is disabled)</li>
</ul>

<h4 id="version-122-2005-06-11">Version 1.2.2 (2005-06-11)</h4>

<h5 id="new-features-6">New features:</h5>
<ul>
  <li>Ability to set the reference genome</li>
  <li>Displays gene name or locus tag for annotated features</li>
</ul>

<h5 id="bug-fixes-1">Bug fixes:</h5>
<ul>
  <li>Uses less memory for MUSCLE alignments by restricting the size of segments aligned by MUSCLE</li>
  <li>Alignment editor shows original sequence coordinates when an aligned character is clicked</li>
  <li>Handles GenBank files with a large number of sequence entries (e.g. C. hominis)</li>
</ul>

<h4 id="version-121-2005-05-05">Version 1.2.1 (2005-05-05)</h4>

<h5 id="new-features-7">New features:</h5>
<ul>
  <li>Experimental support for alignment editing using Cinema-MX Right click on an LCB to trigger the edit menu</li>
  <li>Annotated features are shown in both forward and reverse complement orientation</li>
  <li>When clicking an annotated feature, a database xref popup menu shows options for viewing the feature in known databases. Currently supported databases are Entrez Protein and ASAPdb. Please contact us about support for other databases.</li>
  <li>Choice of either MUSCLE or Clustal in the GUI or on the command line with the –gapped-aligner switch</li>
</ul>

<h5 id="bug-fixes-2">Bug fixes:</h5>
<ul>
  <li>Database xref speed improvement</li>
  <li>Features not shown when viewing large regions to improve speed</li>
  <li>Fixed an off-by-one bug that crashed the viewer when loading Salmonella alignments</li>
  <li>Eliminated extraneous MUSCLE warning messages on the Mac OS X console.</li>
</ul>

<h5 id="known-issues-3">Known issues:</h5>
<ul>
  <li>In some cases MUSCLE on Windows can consume lots of system resources. If you are experiencing problems we suggest using Clustal for the time being.</li>
</ul>

<h4 id="version-120-2005-04-20">Version 1.2.0 (2005-04-20)</h4>

<h5 id="new-features-8">New features:</h5>
<ul>
  <li>Image export. Mauve can export the current display view in JPEG, TIF, and PNG formats at a user-defined resolution.</li>
  <li>MUSCLE Alignment. Mauve now calculates gapped alignments using the MUSCLE DNA alignment method. Previously Clustal-W was used to calculate gapped alignments. MUSCLE provides significantly improved speed and accuracy over Clustal-W</li>
  <li>Collinear genomes. When it is known that the genomes being aligned have no rearrangements, select the Assume collinear genomes option in the “Align sequences” window to tell Mauve that it should find the single best LCB. On the command line, this behavior can be invoked with the –collinear switch.</li>
</ul>

<h5 id="bug-fixes-3">Bug fixes:</h5>
<ul>
  <li>Fixed a memory leak in LCB extension that could quickly consume all system memory.</li>
  <li>The 1.1.0 Windows release used a large amount of memory because it was compiled for use with 160 sequences instead of the usual 16.</li>
  <li>Fixed a bug parsing GenBank sequences which had extra newlines in them</li>
</ul>

<h5 id="known-issues-4">Known issues:</h5>
<ul>
  <li>On Mac OS X the muscle aligner generates lots of warning messages about not being able to open /proc/meminfo. These messages can be ignored and will be eliminated in future releases.</li>
</ul>

<h4 id="version-110-2005-03-30">Version 1.1.0 (2005-03-30)</h4>

<h5 id="new-features-9">New features:</h5>
<ul>
  <li>Mauve can launch a web browser to show a web database entry for annotated features that have a /db_xref qualifier in their GenBank entry. Currently ASAPdb and NCBI’s PubGene databases are supported. Other databases may be added upon request.</li>
  <li>Mauve bundles a text console display for informational and error messages. It can be viewed by selecting Show console from the Help menu.</li>
  <li>Alignment generation has been multithreaded so that alignments can be generated while another alignment is being viewed.</li>
  <li>A Print Preview and Page Layout dialog boxes provide greater control over printing.</li>
</ul>

<h5 id="bug-fixes-4">Bug fixes:</h5>
<ul>
  <li>A bug in mauveAligner which resulting in a “bad alignment” error message has been fixed.</li>
  <li>Printing has been fixed to provide accurate high resolution images. Output is vectorized so images printed to PDF documents (e.g. using Acrobat or PDF995) can be easily embedded in publications and have a high quality look.</li>
  <li>Screen space is used more economically when viewing a large number of sequences, allowing at least six genomes to display on a 1024x768 screen. Previously only three genomes would fit on a screen of that size.</li>
  <li>additional fixed bugs can be found in the Mantis tracker</li>
</ul>

<h4 id="version-101-2005-03-08">Version 1.0.1 (2005-03-08)</h4>

<h5 id="bug-fixes-5">Bug fixes:</h5>
<ul>
  <li>Unable to align genomes on Mac OS X 10.3 (due to a missing library)</li>
  <li>Unable to run Mauve under Linux (Mauve was not set as executable)</li>
  <li>Unable to load genome annotations after constructing an alignment when the source GenBank files reside in a different directory than the alignment output</li>
  <li>Lower memory profile. Mauve now uses less memory when reading alignments, making the large Drosophila alignments available on the web site viewable on computers with over 1GB RAM.</li>
</ul>

<h4 id="version-10-2005-03-01">Version 1.0 (2005-03-01)</h4>

<h5 id="new-features-10">New features:</h5>
<ul>
  <li>Display of annotated features such as CDS, rRNA, tRNA and misc_features from GenBank format files</li>
  <li>Full support for Mac OS X</li>
  <li>Display of chromosomal boundaries and contigs of incomplete genomes</li>
  <li>Display of contig/chromosome name and sequence coordinates</li>
  <li>Source code released under the GNU General Public License. Mauve is now free software and can be modified and redistributed freely (in accordance with the GPL).</li>
</ul>

<h4 id="2004-12-16">2004-12-16</h4>

<h5 id="bug-fixes-6">Bug fixes:</h5>
<ul>
  <li>Inter-LCB islands were (still) incorrect and are now fixed. This bug was originally reported by Val Burland in Dec. 2003 but didn’t get completely fixed until now.</li>
  <li>Vertical only display resize fixed</li>
  <li>Fixed a bug in the LCB info display when the LCB slider had been adjusted</li>
</ul>

<h5 id="new-features-11">New Features:</h5>
<ul>
  <li>Sequence similarity browsing in the display</li>
  <li>More sensitive alignments by using inexact seeds. This means Mauve can be used on more taxa simultaneously and also on more divergent taxa.</li>
  <li>Display alignment in the similarity browser can be accomplished with a single mouse click</li>
</ul>

<h4 id="2004-07-08">2004-07-08</h4>

<h5 id="bug-fixes-7">Bug fixes:</h5>
<ul>
  <li>Brought PDF documentation up-to-date</li>
</ul>

<h4 id="2004-06-30">2004-06-30</h4>

<h5 id="bug-fixes-8">Bug fixes:</h5>
<ul>
  <li>Fixed several missing icon bugs in linux and windows releases</li>
  <li>Mauve now requires Java 1.4 to function correctly, a warning message is displayed if Mauve detects an older version of Java</li>
</ul>

<h5 id="new-features-12">New features:</h5>
<ul>
  <li>Automatic checking for new releases, and on Windows, downloading the new release and starting the installer</li>
  <li>Releases now include a User Guide in PDF format and a link to the online guide</li>
</ul>

<h4 id="2004-05-18">2004-05-18</h4>

<h5 id="bug-fixes-9">Bug fixes:</h5>
<ul>
  <li>Fixed a bug that would result in unreasonably large print jobs</li>
</ul>

<h5 id="new-features-13">New features:</h5>
<ul>
  <li>Implemented interactive LCB weight adjustment. This feature allows dynamic adjustment of the LCB weight within Mauve using a slider control. As the minimum LCB weight is changed the display updates with the corresponding set of LCBs.</li>
  <li>Added toolbar buttons for basic navigational controls such as scrolling and zooming the display. Also added a color mode selector.</li>
  <li>Added keyboard shortcuts for open (ctrl+o), close (ctrl+w), print (ctrl+p), and quit (ctrl+q).</li>
</ul>

<h5 id="important-changes">Important changes:</h5>
<ul>
  <li>The up and down arrows no longer zoom in and out, instead use ctrl+up and ctrl+down</li>
</ul>

<h4 id="2004-04-26">2004-04-26</h4>

<h5 id="bug-fixes-10">Bug fixes:</h5>
<ul>
  <li>Fixed a bug that allowed users to zoom in too far</li>
  <li>Fixed a bug that kept the screen from clearing after selecting close from the File menu</li>
  <li>Fixed a bug that would cause the display to become unaligned after using the context menu to align the display and subsequently zooming out.</li>
  <li>Fixed a bug that raised an exception when no matching regions were found among the sequences being aligned</li>
  <li>Changed the .txt file extension to default to FastA file format instead of raw sequence data</li>
  <li>Fixed a subtle bug that would occasionally cause a “horrible error” to be reported</li>
</ul>

<h5 id="new-features-14">New features:</h5>
<ul>
  <li>Improved the Align sequences dialog interface to give the user an easy set of default values with optional control over the alignment parameters. Also added an interface to set the backbone parameters.</li>
  <li>Added the ability to change ordering of genomes in the display using drag and drop. To enter and leave reordering mode press the h key. When in reordering mode genomes can be dragged and dropped into the desired order.</li>
</ul>

<h4 id="2004-02-19">2004-02-19</h4>

<h5 id="bug-fixes-11">Bug fixes:</h5>
<ul>
  <li>Fixed a bug in the island coordinate output that caused the end positions of inverted regions to be huge (wrong) numbers. Removed obsolete -d and –stats command line options. Added a –id-matrix or -t command line option to print out an identity matrix. Mauve now creates an identity matrix by default.</li>
</ul>

<h4 id="2003-12-09">2003-12-09</h4>

<h5 id="bug-fixes-12">Bug fixes:</h5>
<ul>
  <li>Fixed another bug in the LCB extension code that occasionally resulted in incorrect alignments.</li>
</ul>

<h4 id="2003-12-06">2003-12-06</h4>

<h5 id="bug-fixes-13">Bug fixes:</h5>
<ul>
  <li>Fixed a serious bug that caused the first sequence to be misaligned in regions with numerous gaps.</li>
</ul>

<h4 id="2003-11-04">2003-11-04</h4>

<h5 id="bug-fixes-14">Bug fixes:</h5>
<ul>
  <li>Fixed a minor file format problem with XMFA (eXtended Multi-FastA) output. Fixed a bug with reverse complement backbone determination.</li>
</ul>

<h4 id="2003-10-17">2003-10-17</h4>

<h5 id="bug-fixes-15">Bug fixes:</h5>
<ul>
  <li>Updated the GUI interface to match new command line options for the aligner. The GUI was behaving badly in the 2003-10-08 release. Full alignment is now selected by default, if no LCB weight is specified one will be chosen automatically.</li>
</ul>

<h4 id="2003-10-08">2003-10-08</h4>

<h5 id="bug-fixes-16">Bug fixes:</h5>
<ul>
  <li>Final aligner debugging completed using thousands of test cases. Numerous bugs were squashed. The aligner should be (but isn’t) rock solid now. Removed file I/O from the ClustalW code, resulting in vast speedups during recursive alignment. One more bug to kill.</li>
</ul>

<h4 id="2003-09-13">2003-09-13</h4>

<h5 id="bug-fixes-17">Bug fixes:</h5>
<ul>
  <li>Debugged the LCB extension code, hopefully eliminating strange crashes from the previous release.</li>
</ul>

<h4 id="2003-09-08">2003-09-08</h4>

<h5 id="new-features-15">New Features:</h5>
<ul>
  <li>Added LCB extension code, allows Mauve to search for additional MUM anchors off the ends of LCBs. Increased the minimum anchor size to 5 b.p. to reduce spurious anchoring.</li>
</ul>

<h4 id="2003-08-15">2003-08-15</h4>

<h5 id="new-features-16">New Features:</h5>
<ul>
  <li>Added print menu item to the File menu - Clean-ups for Linux - Mauve for Linux is ready!</li>
</ul>

<h4 id="2003-08-13">2003-08-13</h4>

<h5 id="bug-fixes-18">Bug fixes:</h5>
<ul>
  <li>Additional bugfixes for the recursive anchoring and alignment process. Will this software ever be bug free?</li>
</ul>

<h4 id="2003-08-02">2003-08-02</h4>

<h5 id="bug-fixes-19">Bug fixes:</h5>
<ul>
  <li>Fixed some serious bugs in the production of gapped alignments from Mauve style MUM alignments. This bug affected island predictions and backbone predictions of previous releases.</li>
</ul>

<h5 id="new-features-17">New features:</h5>
<ul>
  <li>Implemented Multi-FastA sequence input. If a single sequence file is given Mauve assumes it is a Multi-FastA file where each sequence entry corresponds to a sequence to align.</li>
</ul>

<h4 id="2003-07-31">2003-07-31</h4>

<h5 id="new-features-18">New features:</h5>
<ul>
  <li>Implemented backbone identification – backbone is any region of sequence conserved among all genomes under study. Mauve’s definition of backbone is any region longer than x base pairs that does not contain any gaps longer than x base pairs. x is specified as the island size in the “Align Sequences…” window.</li>
</ul>

<h4 id="2003-07-30">2003-07-30</h4>

<h5 id="bug-fixes-20">Bug fixes:</h5>
<ul>
  <li>Fixed a segfault in SML binary search</li>
</ul>

<h5 id="new-features-19">New features:</h5>
<ul>
  <li>Added end-gap penalties to clustal alignments</li>
</ul>

<h4 id="2003-07-28">2003-07-28</h4>

<h5 id="bug-fixes-21">Bug fixes:</h5>
<ul>
  <li>Fixed a few problems with Clustal Alignment – more to come</li>
</ul>

<h4 id="2003-07-25">2003-07-25</h4>

<h5 id="bug-fixes-22">Bug fixes:</h5>
<ul>
  <li>Changed the default maximum Clustal alignment size to 9000 - Fixed numerous bugs related to recursive alignment and ClustalW alignment</li>
</ul>

<h5 id="new-features-20">New features:</h5>
<ul>
  <li>Now outputs alignments in Extended Multi-FastA (xmfa) format</li>
</ul>

<h4 id="2003-06-12">2003-06-12</h4>

<h5 id="bug-fixes-23">Bug fixes:</h5>
<ul>
  <li>Don’t write a gapped alignment unless the user requests a Full alignment</li>
</ul>

<h4 id="2003-05-08">2003-05-08</h4>

<h5 id="bug-fixes-24">Bug fixes:</h5>
<ul>
  <li>bugfixes for displaying large ( &gt; 2 G.B.) sequences</li>
</ul>

<h5 id="new-features-21">New features:</h5>
<ul>
  <li>new ruler label style for easier reading</li>
</ul>

<h4 id="2003-05-05">2003-05-05</h4>

<h5 id="bug-fixes-25">Bug fixes:</h5>
<ul>
  <li>Fixed DragNDrop for align sequences window</li>
</ul>

<h4 id="2003-04-28">2003-04-28</h4>

<h5 id="bug-fixes-26">Bug fixes:</h5>
<ul>
  <li>Fixed Align sequences window disappearance</li>
</ul>

<h5 id="new-features-22">New features:</h5>
<ul>
  <li>Added clustalW format alignment output (command line mauveAligner only)</li>
</ul>

<h4 id="2003-04-26">2003-04-26</h4>

<h5 id="bug-fixes-27">Bug fixes:</h5>
<ul>
  <li>Fixed printing (broken in previous release)</li>
</ul>

<h5 id="new-features-23">New features:</h5>
<ul>
  <li>Supports drag and drop of alignments to the alignment window and of sequence files to the sequences list</li>
</ul>

<h4 id="2003-04-21">2003-04-21</h4>

<h5 id="bug-fixes-28">Bug fixes:</h5>
<ul>
  <li>Fix mouse click accuracy problem when zoomed in</li>
  <li>don’t clear alignment window after alignment</li>
</ul>

<h5 id="new-features-24">New features:</h5>
<ul>
  <li>mauveAligner finds interLCB islands</li>
  <li>copy the File chooser view to new alignment windows</li>
</ul>

<h4 id="2003-04-04">2003-04-04</h4>

<h5 id="new-features-25">New features:</h5>
<ul>
  <li>Have mauveAligner output islands, tree, and alignment</li>
  <li>Implement high-resolution printing hack</li>
</ul>

<h4 id="2003-04-03">2003-04-03</h4>

<h5 id="new-features-26">New features:</h5>
<ul>
  <li>Add LCB color lines to link LCBs</li>
</ul>

<h4 id="2003-02-21">2003-02-21</h4>

<h5 id="new-features-27">New features:</h5>
<ul>
  <li>Initial Release - Packaged into an installer for windows</li>
</ul>

  </article>

</div>


  </td>
  </table>



      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li>the Darling lab | computational (meta)genomics</li>
          <li><a href="mailto:aaron.darling@uts.edu.au">aaron.darling@uts.edu.au</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/koadman">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          

          
          <li>
            <a href="https://twitter.com/koadman">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
</p>
      </div>
    </div>
    <img alt="tree of bacterial and archaeal life, showing human gut lineages in blue" src="/platypusfarm//assets/tree.png">

  </div>
</footer>


  </body>

</html>
