<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width initial-scale=1">

  <title>Software</title>
  <meta name="description" content="The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
">

  <link rel="stylesheet" href="/platypusfarm/css/main.css">
  <link rel="canonical" href="https://koadman.github.io/platypusfarm/software/">
  <link rel="alternate" type="application/atom+xml" title="the Darling lab | computational (meta)genomics" href="https://koadman.github.io/platypusfarm/feed.xml" />
  <link rel="shortcut icon" href="/platypusfarm/site32.jpg" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/platypusfarm/">the Darling lab | computational (meta)genomics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/mauve/mauve.html">Mauve</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/openings/">Openings</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/people/">People</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/projects/">Projects</a>
          
        
          
          <a class="page-link" href="/platypusfarm/publications/">Publications</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/reviews/">Reviews</a>
          
        
          
        
          
          <a class="page-link" href="/platypusfarm/software/">Software</a>
          
        
          
        
          
        
          
        
	<!-- special link for tutorials pages, which are done with MkDocs -->
        <a class="page-link" href="/platypusfarm/tutorials">Tutorials</a>
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Software</h1>
  </header>

  <article class="post-content">
    <h3 id="past-software-projects">Past software projects</h3>

<p>Darling lab members have contributed to the following software projects in the past but are no longer actively participating in these projects.</p>

<p><a href="http://darlinglab.org/mauve/mauve.html"><strong>Mauve</strong> - multiple genome alignment</a><br />
Mauve is a software tool to compute whole genome multiple alignments among bacteria and small eukaryotic genomes (usually no bigger than Drosophila). The software includes a Java-based visualization module and a set of alignment programs written in C++, available for Linux, Mac, and Windows. Source code is currently available on <a href="http://sf.net/p/mauve">sourceforge.net</a>.</p>

<p><a href="http://phylosift.wordpress.org"><strong>PhyloSift</strong> - phylogenetic analysis of genomes and metagenomes</a><br />
PhyloSift is a re-implementation and extension of the original amphora approach to reconstructing metagenome phylogeny and taxonomy. PhyloSift has several new capabilities, including an ability to quickly process reads generated by next-generation sequencers, an expanded set of phylogenetic marker genes that includes the Archaea, Eukarya, and Viruses, and Bayesian phylogenetic placement of reads using the <a href="http://matsen.fhcrc.org/pplacer">pplacer software</a>.</p>

<p><a href="http://sf.net/p/ngopt"><strong>A5-miseq</strong> - Andrew And Aaron’s Awesome Assembly Pipeline (miseq enabled)</a><br />
A5-miseq is an updated software pipeline for de novo assembly of microbial isolate genomes from Illumina HiSeq or MiSeq data. It automates all of the usual steps in genome assembly with an easy-to-use interface.</p>

<p><a href="https://github.com/beagle-dev/beagle-lib"><strong>The beagle library</strong> - a library for phylogenetic likelihood calculation on the CPU and GPU</a><br />
libhmsbeagle, also known as the beagle library, is a software development library for calculating phylogenetic likelihoods using a variety of compute hardware types. Currently supported hardware types and features include graphics processing units (GPUs) via CUDA and OpenCL, standard CPUs using SSE for fine-scale parallelism and OpenMP for coarse parallelism, and others. The library works on Linux, Mac, and Windows. Further development on this software is being led by [Daniel Ayres].</p>

<p><a href="https://github.com/OnlinePhylo/sts"><strong>STS</strong> - the Sequential Tree Sampler</a><br />
Software to update a posterior probability distribution over phylogenies with new sequence data. A collaboration with the <a href="http://matsen.fhcrc.org">Matsen group</a> at the Fred Hutchinson Cancer Research Center.</p>

<p><a href="http://mpiblast.org/"><strong>mpiBLAST</strong> - open source parallel BLAST</a><br />
mpiBLAST is a parallelization of the popular NCBI BLAST for MPI-based compute clusters. When searching large databases, it can yield super-linear speedups. It is extremely flexible, accommodating cluster architectures with and without shared storage and parallel filesystems. It integrates well with most job scheduling systems and has also been extended to grid architectures.</p>

<p><a href="http://wwwabi.snv.jussieu.fr/public/Repeatoire/"><strong>Repeatoire</strong> – alignment of interspersed genomic repeats</a><br />
Software to construct multiple sequence alignments of interspersed repeats directly from raw genomic sequence. This project is led by <a href="http://www.cbcb.umd.edu/~treangen/">Dr. Todd Treangen</a>.</p>

<p><a href="https://github.com/xavierdidelot/clonalorigin"><strong>ClonalOrigin</strong> – inference of homologous recombination in bacteria using whole genome sequences</a><br />
This software implements a Bayesian model of recombination among bacteria. Starting with whole genome sequences aligned with progressiveMauve, this software can reconstruct the probability distribution over historical recombination events among each lineage. From that probability distribution one can compute quantities of interest, such as the rate of gene flow between particular lineages, which parts of the genome have undergone recombination, and more. We have applied this software to investigate speciation in archaea, the chromosomal structure of recombination in Bacillus, and rates of gene flow between pathogenic and commensal E. coli. This software is being maintained by <a href="http://www.xavierdidelot.xtreemhost.com/">Xavier Didelot</a>.</p>

<p><a href="http://darlinglab.org/barphlye"><strong>barphlye</strong> – Bayesian rearrangement phylogeny in Yersinia</a><br />
barphlye supplements the <a href="http://badger.duq.edu/">BADGER software</a> to analyze patterns present in ancient genome arrangements. A modified version of BADGER samples reconstructions of inversion phylogeny among a set of related organisms, and barphlye can then detect bias in the reconstructed ancestral genome arrangements. Using barphlye, I was able to discover rearrangement hotspots near the origin of replication in bacterial chromosomes. I was also able to confirm the bias towards “symmetric” inversion in circular bacterial chromosomes. Although Yersinia is in the title, the software can be applied to any bacteria with circular chromosomes, and can even operate on individual linear chromosomes.</p>

<p><a href="http://www.xavierdidelot.xtreemhost.com/genoplast.htm"><strong>GenoPlast</strong> – Bayesian inference of genomic plasticity</a><br />
Given a Mauve genome alignment, GenoPlast uses a statistical model to infer the baseline rates of gene gain and loss among a group of organisms, along with lineage-specific changes to those rates. Gain and loss are modeled independently. Thus it is possible to detect, for example, a lineage-specific accelerated loss with a constant rate of acquisition that may be characteristic of a recent lifestyle change in bacteria. This project was led by <a href="http://www.xavierdidelot.xtreemhost.com/">Xavier Didelot</a>.</p>

<p><a href="http://seevolution.sourceforge.net/"><strong>Seevolution</strong> – a time machine for evolution</a><br />
Seevolution is an interactive viewer for mutations occurring during genome evolution. The program is written in Java and uses Java3D. This project was developed by Andres Esteban-Marcos, who was a student I supervised at The Institute for Molecular Bioscience at The University of Queensland.</p>

<p><a href="http://asap.ahabs.wisc.edu/asap/home.php"><strong>ASAP and the ERIC BRC</strong></a><br />
ASAP is A Systematic Annotation Package for microbial genomes which is part of the larger Enteropathogen Resource Integration Center project. Together, ERIC and ASAP provide a centralized, web based means to annotate the genomes of enteropathogens and serve as a clearinghouse for all types of annotation and experimental data. ASAP supports a wealth of automated annotation strategies, and its evolution has been described in a series of Nucleic Acids Research papers. ASAP is led by <a href="http://genetics.wisc.edu/Perna.htm">Nicole Perna</a> and continues to grow at the Genome Evolution Laboratory since my departure.</p>

<p><a href="https://sourceforge.net/p/mauve/code/HEAD/tree/libClustalW"><strong>libClustalW</strong> – a C library for Clustal-W 1.83</a><br />
The 1.83 release of Clustal-W has been refactored as a C library. It builds in Visual Studio on Windows and on Linux, BSD, and other unices with gcc and automake. Note that the Clustal-W authors have since <a href="http://www.clustal.org/">made a 2.0 release</a> which represents a complete rewrite of the aligner and which renders this library obsolete.</p>

<p><a href="http://sourceforge.net/projects/libgenome"><strong>libGenome</strong> – a C++ development library</a><br />
libGenome is an open-source C++ library for reading and writing genome sequence data from common file formats, and also provides functions for basic manipulation of genome sequences. It was designed from the ground-up for speed and efficiency. It is available on sourceforge and also as part of the debian linux distribution.</p>


  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li>the Darling lab | computational (meta)genomics</li>
          <li><a href="mailto:aaron.darling@uts.edu.au">aaron.darling@uts.edu.au</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/koadman">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          

          
          <li>
            <a href="https://twitter.com/koadman">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">koadman</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">The Darling lab at the University of Technology Sydney. We develop computational and molecular techniques to characterize the hidden world of microbes.
</p>
      </div>
    </div>
    <img alt="tree of bacterial and archaeal life, showing human gut lineages in blue" src="/platypusfarm/assets/tree.png">

  </div>
</footer>


  </body>

</html>
