---
layout: page
title: Projects
permalink: /projects/
---

Our group has several ongoing projects.

+ Bayesian phylogenomic inference using Monte Carlo methods, including Sequential Monte Carlo, MCMC, and approximate Bayesian computation. See the [STS software](https://github.com/OnlinePhylo/sts) and [beagle library](https://github.com/beagle-dev/beagle-lib) for examples of recent work in this area.
+ Understanding the development of the infant gut microbiome, the relationship with maternal microbiota, and the development of atopy.
  In collaboration with a large team led by [Prof. Shyamali Dharmage](https://findanexpert.unimelb.edu.au/display/person3474) and with support of the National Health and Medical Research Council, we are studying the microbiota found in the breast milk of families that are at risk for atopy.
+ Forensic analysis of microbial samples using phylogenetic analysis of metagenome data.
  In collaboration with the labs of [Jonathan Eisen at UC Davis](http://phylogenomics.wordpress.com) and [Erick Matsen at FHCRC](http://matsen.fhcrc.org) we develop the [PhyloSift software](http://phylosift.wordpress.org), which provides an integrated set of tools for Bayesian phylogenetic & ecological analysis of metagenomic data.
+ Development of advanced techniques for studying microbial genomes and metagenomes by tightly coupled development of computational inference methods and next-generation sequencing techniques. Two examples of recent work in this area are the development of [metagenomic Hi-C](https://peerj.com/articles/415/) and the development of [full length 16S ribosomal gene sequencing](http://biorxiv.org/content/early/2014/11/06/010967) on the Illumina MiSeq.
+ Comparative genomics and genome assembly. We are developing and applying wet-lab protocols and software pipelines to sequence the genomes of microbial isolates _en masse_ using the Oxford Nanopore, Pacific Biosciences, and Illumina platforms. Recent work in this area includes the [A5-miseq](http://sf.net/p/ngopt) assembly pipeline.
